subprojects/gdk_rust/gdk_common/src/scripts.rs

```diff
@@ -21,3 +35,51 @@ pub fn p2shwpkh_script_sig(public_key: &PublicKey) -> Script {
         .into_script();
     Builder::new().push_slice(internal.as_bytes()).into_script()
 }
+
+impl ScriptType {
+    pub fn types() -> &'static [ScriptType] {
+        &TYPES
+    }
+
+    pub fn first_account_num(self) -> u32 {
+        self as u32
+    }
+
+    pub fn is_segwit(self) -> bool {
+        matches!(self, ScriptType::P2wpkh | ScriptType::P2shP2wpkh)
+    }
+
+    /// Returns a mock witness with the expected size
+    pub fn mock_witness(self) -> Vec<Vec<u8>> {
+        match self {
+            // signature (72) + compressed public key (33)
+            ScriptType::P2wpkh | ScriptType::P2shP2wpkh => vec![vec![0u8; 72], vec![0u8; 33]],
+            // empty for non-witness inputs
+            ScriptType::P2pkh => vec![],
+        }
+    }
+
+    /// Returns a mock script sig with the expected size
+    pub fn mock_script_sig(self) -> Vec<u8> {
+        match self {
+            // empty for native segwit
+            ScriptType::P2wpkh => vec![],
+            // OP_PUSHBYTES <22 bytes>
+            ScriptType::P2shP2wpkh => vec![0u8; 23],
+            // OP_PUSHBYTES <72 bytes sig> OP_PUSHBYTES <33 bytes compressed key>
+            ScriptType::P2pkh => vec![0u8; 107],
+        }
+    }
+
+    /// Returns a mock scriptPubkey with the expected size
+    pub fn mock_script_pubkey(self) -> Vec<u8> {
+        match self {
+            // OP_0 PUSHBYTES <20 bytes hash>
+            ScriptType::P2wpkh => vec![0u8; 22],
+            // OP_HASH160 PUSHBYTES <20 bytes hash> OP_EQUAL
+            ScriptType::P2shP2wpkh => vec![0u8; 23],
+            // OP_DUP OP_HASH160 OP_PUSHBYTES_20 <20 bytes hash> OP_EQUALVERIFY OP_CHECKSIG
```

Comment:
nit: in `mock_script_*`,  use one of `OP_PUSHBYTES`, `PUSHBYTES`, `OP_PUSHBYTES_n` in comments

Reply:
Addressed by https://github.com/Blockstream/gdk/pull/93/commits/1d84f184b7b94a96e17161641e19c5b49cd4f5f7.